server/channels/jobs/user_deletion/worker.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package user_deletion
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+	"github.com/mattermost/mattermost/server/v8/channels/jobs"
+)
+
+const jobName = "UserDeletion"
+
+type AppIface interface {
+	GetUser(userID string) (*model.User, *model.AppError)
+	PermanentDeleteUser(rctx request.CTX, user *model.User) *model.AppError
+}
+
+func MakeWorker(jobServer *jobs.JobServer, app AppIface) *jobs.SimpleWorker {
```

Comment:
I guess the main question I have is: at the moment this isn't really a "job", it's more like a deferred api call. If we wanted to make it more job-like, we could chunk it, and update the progress, and then it would be resumable. True, it would be more work, but the end result would be something that we could use as a model for other long running tasks too.

Reply:
For now, it is exactly a deferred API call. And the objective is to eventually turn it to a proper job. Is your suggestion to do the proper job bit as part of this PR itself?

Reply:
If fixing this is not urgent, then I'd agree with Chris. Implementing a proper job seems like the way to go, and queing it for v11 would also give us room for breaking changes (if needed).

Reply:
Ok I'll keep chipping away at this, while changing the approach to spawn a job from mmctl as discussed here: https://github.com/mattermost/mattermost/pull/30969#discussion_r2095716162. This would mean we can merge it anytime we are done without being blocked by anything.